### DISCUSSION TITLE
Forking Messages and Conversations is very difficult to understand

### DISCUSSION THREAD

**Discussion by sunsky89757** [danny-avila/LibreChat#2908] [Q&A]
### What is your question?

![image](https://github.com/danny-avila/LibreChat/assets/113254524/4cc5d98a-1961-4936-9d5f-6de8e49375af)

### More Details

I'm very sorry, I think my friends and I have read this part of your feature documentation, but we still don't understand how to use it. From a user perspective, I think most people will find this feature confusing. Could you please provide a clearer explanation or example scenarios to illustrate it? Thank you.

### What is the main subject of your question?

Documentation

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Have you tried it very much? the best way to understand it is to use it. Then you can offer more clarity and submit a PR to the documentation: https://github.com/LibreChat-AI/librechat.ai

**Comment by danny-avila** [comment]
Here's a simple visualization from discord:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/fcc61310-0e3b-4b68-80db-5d9f0d81ddf5)

**Comment by danny-avila** [comment]
This is more of a "power user" feature, for those already accustomed to message branching in general.

**Comment by sunsky89757** [comment]
![image](https://github.com/danny-avila/LibreChat/assets/113254524/520df09e-a9a1-4862-9cb0-15b645d21b05)
I think this is the key issue, the regenerated content is called a branch. The three choices of a new conversation forked out are the process of selecting the number of answers for this branch.

**Reply by danny-avila** [reply]
Yes, each regeneration is a branch. You can branch from the root message too by editing it.

EDIT: Actually, given that there can be multiple root nodes, it is technically a "forest."

**Reply by danny-avila** [reply]
This data structure is called a [Generic Tree or N-ary Tree,](https://www.geeksforgeeks.org/generic-treesn-array-trees/), part of the complexity comes from representing them in 2-dimensional, linear way as is done here and on ChatGPT.

**Comment by sunsky89757** [comment]
Thank you, I've gained more knowledge, haha!
